popx/transaction_test.go

```diff
@@ -130,3 +143,28 @@ func (t popWriteSkewTest) UpdateBalance(
 	}
 	return nil
 }
+
+func collectCount(t *testing.T) (labelName, labelValue string, count int) {
+	var (
+		mChan   = make(chan prometheus.Metric)
+		metrics []prometheus.Metric
+		done    = make(chan struct{})
+	)
+	go func() {
+		defer close(done)
+		for m := range mChan {
+			metrics = append(metrics, m)
+		}
+	}()
+	TransactionRetries.Collect(mChan)
+	close(mChan)
```

Comment:
Closing a channel as the receiver? This looks fishy :sweat_smile:
Why not just range over the channel in this function instead of the go routine? Also this seems racy that you close the channel while `.Collect` might still write to it :thinking:

Reply:
This is lifted, with modifications, from the prometheus/testutil package. It works, so 🤷 😁